pkg/ui/src/views/reports/containers/range/index.tsx

```diff
@@ -158,20 +171,23 @@ class Range extends React.Component<RangeProps, {}> {
           rangeID={responseRangeID}
           log={this.props.rangeLog}
           lastError={this.props.rangeLogLastError}
+          inFlight={this.props.rangeLogInFlight}
         />
       </div>
     );
   }
 }
 
-function mapStateToProps(state: AdminUIState) {
+function mapStateToProps(state: AdminUIState, props: RangeProps) {
+  const rangeID = props.params[rangeIDAttr];
   return {
-    range: state.cachedData.range.data,
-    allocatorRange: state.cachedData.allocatorRange.data,
-    rangeLog: state.cachedData.rangeLog.data,
-    lastError: state.cachedData.range.lastError,
-    allocatorLastError: state.cachedData.allocatorRange.lastError,
-    rangeLogLastError: state.cachedData.rangeLog.lastError,
+    range: state.cachedData.range[rangeID] && state.cachedData.range[rangeID].data,
+    allocatorRange: state.cachedData.allocatorRange[rangeID] && state.cachedData.allocatorRange[rangeID].data,
+    rangeLog: state.cachedData.rangeLog[rangeID] && state.cachedData.rangeLog[rangeID].data,
+    lastError: state.cachedData.range[rangeID] && state.cachedData.range[rangeID].lastError,
+    allocatorLastError: state.cachedData.allocatorRange[rangeID] && state.cachedData.allocatorRange[rangeID].lastError,
+    rangeLogLastError: state.cachedData.rangeLog[rangeID] && state.cachedData.rangeLog[rangeID].lastError,
+    rangeLogInFlight: state.cachedData.rangeLog[rangeID] && state.cachedData.rangeLog[rangeID].inFlight,
```

Comment:
I'd prefer we pass the `CachedDataReducer` state to each, as opposed to three variables for each (the data, `lastError`, and `inFlight`)